File: BilDunya/src/main/java/com/bildunya/dto/BadgeDto.java
package com.bildunya.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BadgeDto {

    /** Material Symbols adı: public, castle, photo_camera, lock */
    private String icon;

    private String title;

    private String subtitle;

    @JsonProperty("unlocked")
    private boolean unlocked;
}

File: bildunya_frontend/lib/data/models/badge_dto.dart
/// [com.bildunya.dto.BadgeDto]
class BadgeDto {
  const BadgeDto({
    this.icon,
    this.title,
    this.subtitle,
    this.unlocked = false,
  });

  final String? icon;
  final String? title;
  final String? subtitle;
  final bool unlocked;

  factory BadgeDto.fromJson(Map<String, dynamic> json) {
    return BadgeDto(
      icon: json['icon'] as String?,
      title: json['title'] as String?,
      subtitle: json['subtitle'] as String?,
      unlocked: json['unlocked'] as bool? ?? false,
    );
  }
}
